Core/Core/Models/AssigmentGroup.swift

```diff
@@ -37,15 +36,9 @@ public final class AssignmentGroup: NSManagedObject {
         model.position = item.position
         model.courseID = courseID
 
-        if let apiAssignments = item.assignments, apiAssignments.count > 0 {
-            model.assignments = Set<Assignment>()
-            for a in apiAssignments {
-                let assignmentPred = NSPredicate(format: "%K == %@", #keyPath(Assignment.id), a.id.value)
-                let dbModel: Assignment = context.fetch(assignmentPred).first ?? context.insert()
-                dbModel.update(fromApiModel: a, in: context, updateSubmission: false)
-                model.assignments.insert(dbModel)
-            }
-        }
+        let assignmentPredicate = NSPredicate(format: "%K == %@", #keyPath(Assignment.assignmentGroupID), item.id.value)
+        let assignments: [Assignment] = context.fetch(assignmentPredicate)
+        assignments.forEach { $0.assignmentGroupPosition = item.position }
```

Comment:
this is the only way to get the sections in the right order

Reply:
I think this is fine.

Reply:
In `Assignment.save` you could look up its AssignmentGroup and update the `assignmentGroupPosition`. That way they don't have to be requested and saved in a particular order.